src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6919,7 +6919,8 @@ static bool dependsOnTypeParameter(TypeParameterSymbol typeParameter1, TypeParam
             Debug.Assert(operandType.Type is object);
 
             var parameter = methodOpt.Parameters[0];
-            var parameterType = parameter.TypeWithAnnotations;
+            var parameterAnnotations = GetParameterAnnotations(parameter);
```

Comment:
Looking through this file there are a number of places where we still use `parameter.TypeWithAnnotations` vs. `GetParameterAnnotations`. Why did we need to update this call and have we audited the other uses in this file to see if they would benefit from `GetParameterAnnotations`?

Reply:
Basically if you just access something like the TypeWithAnnotations on a symbol, and you don't try to "fold in" the FlowAnalysisAnnotations on that symbol into the TypeWithAnnotations, you probably have bugs related to nullability attributes. It does feel like there should be something like `GetAdjustedTypeWithAnnotations()` (strawman name) on certain kinds of symbols in order to make it easier to do the right thing.

Reply:
To clarify, I think such a refactoring should be attempted in a separate PR.

Reply:
Let's file a follow up issue to look at this. Just looking around in this file I see lots of instances of the pattern that you fixed here. It's possible those are intentional but if so think it would help to have both a helper method like you suggested plus a declarative method of stating "I have reasons for just acessing TWA directly here".

Reply:
Opened #47635